retriever/lib/datapackage.py

```diff
@@ -405,17 +406,36 @@ def edit_json(json_file):
     try:
         contents = json.load(
             open(os.path.join(HOME_DIR, 'scripts', json_file), 'r'))
-    except FileNotFoundError:
+    except (IOError,OSError):
         print("Script not found.")
         return
 
     edit_dict(contents, 1)
 
     file_name = contents['name'] + ".json"
+    file_name = file_name.replace('-', '_')
     with open(os.path.join(HOME_DIR, 'scripts', file_name), 'w') as output_file:
         json_str = json.dumps(contents, output_file, sort_keys=True, indent=4,
                               separators=(',', ': '))
         output_file.write(json_str + '\n')
         print("\nScript written to " +
               os.path.join(HOME_DIR, 'scripts', file_name))
         output_file.close()
+
+def delete_json(json_file):
+    try:
+        #delete scripts from home directory
+        os.remove(os.path.join(HOME_DIR, 'scripts', json_file))
+        os.remove(os.path.join(HOME_DIR, 'scripts', json_file[:-4] + 'py'))
+        os.remove(os.path.join(HOME_DIR, 'scripts', json_file[:-4] + 'pyc'))
+
+        #delete scripts from current directory if exists
+        os.remove(os.path.join(os.getcwd(), 'scripts', json_file))
```

Comment:
Hi, henry I referred [this](http://stackoverflow.com/questions/10840533/most-pythonic-way-to-delete-a-file-which-may-not-exist) for `os.remove()`. what do you suggest, which will be better?
About using `py*`, smart choice :smiley: I would do that.